### PR TITLE
chore: update dev dependencies

### PR DESCRIPTION
**Pull request by matthewgallo**
This PR was automatically generated to update versions of dev dependencies to their latest versions. This helps ensure we get fixes and improvements in a timely fashion and reduces the impact of updating the versions that can arise if versions are allowed to become stale for an extended period.

#### What did you change?

This action ran `yarn upgrade:automatic` to update `package.json` files to request the latest versions of each package. Certain critical packages are intentionally excluded.

This action also deleted the `yarn.lock` file in order to recreate it with the latest matching versions of secondary dependencies.

This PR includes the various `package.json` that pull our dependencies forward to the latest versions, the `yarn.lock` update that maps required versions to the actual versions to be used, and updates the offline mirror.

#### How did you test and verify your work?

This PR should not normally significantly affect the build outputs or runtime packages. Ensure that `yarn ci-check` runs cleanly and all tests pass (done automatically as part of the PR checks).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,43 +64,43 @@
...
-    "accessibility-checker": "^3.1.65",
+    "@typescript-eslint/eslint-plugin": "^7.9.0",
+    "@typescript-eslint/parser": "^7.9.0",
+    "accessibility-checker": "^3.1.71",
     "cheerio": "^1.0.0-rc.12",
     "copyfiles": "^2.4.1",
-    "cspell": "^8.3.2",
-    "eslint": "^8.56.0",
+    "cspell": "^8.8.1",
+    "eslint": "^9.2.0",
```
Comment (kennylam) [inline package.json:83]: `eslint-config-carbon` is still using v8, which will probably conflict with this upgrade. At a glance, it looks like `.eslintignore` is no longer supported in v9, and must be included in the config file. Suggest leaving this at v8 until it's necessary to upgrade.

File: packages/ibm-products/package.json [hunk 2]
```diff
@@ -54,52 +54,52 @@
...
-    "@rollup/plugin-typescript": "^11.0.0",
+    "@rollup/plugin-babel": "^6.0.4",
+    "@rollup/plugin-commonjs": "^25.0.7",
+    "@rollup/plugin-node-resolve": "^15.2.3",
+    "@rollup/plugin-typescript": "^11.1.6",
     "babel-plugin-dev-expression": "^0.2.3",
     "babel-preset-ibm-cloud-cognitive": "^0.14.40",
     "chalk": "^4.1.2",
-    "change-case": "4.1.2",
+    "change-case": "5.4.4",
```
Comment (kennylam) [inline packages/ibm-products/package.json:74]: I noticed the ci-checks [complaining](https://github.com/carbon-design-system/ibm-products/actions/runs/9089835983/job/24981641534?pr=5228#step:6:87) about a missing export for `paramCase`. It seems in v5 `paramCase` has been renamed to `kebabCase` ([release notes](https://github.com/blakeembrey/change-case/releases/tag/change-case%405.0.0)).

Usage docs have changed as well, although I'm not sure if it requires a rewrite to the existing imports. According to the latest docs, it should look something like

```js
import * as changeCase from 'change-case';

const substitutions = {
  DISPLAY_NAME: changeCase.pascalCase(name),
```

File: packages/ibm-products/package.json [hunk 3]
```diff
@@ -54,52 +54,52 @@
...
   },
   "devDependencies": {
-    "@babel/cli": "^7.23.9",
-    "@babel/core": "^7.23.9",
+    "@babel/cli": "^7.24.5",
+    "@babel/core": "^7.24.5",
     "@babel/plugin-proposal-class-properties": "^7.17.12",
-    "@babel/plugin-proposal-export-default-from": "^7.17.12",
+    "@babel/plugin-proposal-export-default-from": "^7.24.1",
     "@babel/plugin-proposal-export-namespace-from": "^7.17.12",
```
Comment (kennylam) [inline packages/ibm-products/package.json:61]: `@babel/plugin-proposal-class-properties` and `@babel/plugin-proposal-export-namespace-from` have been merged to the standard and are no longer proposals. The graduated packages can be used (sorry it won't let me make a suggestion across deleted lines):

- [@babel/plugin-transform-class-properties](https://www.npmjs.com/package/@babel/plugin-transform-class-properties)
- [@babel/plugin-transform-export-namespace-from](https://www.npmjs.com/package/@babel/plugin-transform-export-namespace-from)

**Review by kennylam — COMMENTED** [review]
@matthewgallo Looks like a few major bumps are affecting the build and/or CI. I've listed some of the ones I could find.

**Comment by matthewgallo** [comment]
Thanks for reviewing @kennylam! Some extra context, these PRs that upgrade dependencies are scheduled to open on the 1st and 15th of every month. It's opened by me because I was the last person to update the action that opens this PR 😄

**Comment by kennylam** [comment]
Ah, gotcha!

**Comment by elycheea** [comment]
Replaced by #5389 ... but we should probably bring this up in our huddle so people know how to address this when if it gets auto-assigned.

**Pull request closed without merging**